Improvements to query docs (#149)
resource-watch/doc-api: source/includes/api/_query.md

format                 | The format of the returned response. By default, JSON format is assumed (`json`), but you can also request the response as CSV (`csv`), in which case the returned response will contain the CSV contents of the response. **This parameter will only be considered for document-based datasets.** | String | No |
freeze                 | The `freeze` parameter, when provided as `true`, will create a file with the results of the execution of the query and return the URL for that file. **Please note that you should be authenticated in order to request freezing the results of query executions.** | Boolean | No |
geostore               | Read more about the geostore query parameter [here](/index-rw.html#filter-query-results-by-geostore). | String | No |

#### Filter query results by geostore

> Example query providing a geostore id as query parameter to filter the results:

```shell
curl -i -X GET 'http://api.resourcewatch.org/v1/query/1d7085f7-11c7-4eaf-a29a-5a4de57d010e?sql=SELECT * FROM dis_001_significant_earthquakes LIMIT 5&geostore=972c24e1da2c2baacc7572ee9501abdc'
```

> Some dataset providers support receiving a `geostore` query parameter. When providing this parameter, you can request geo-referenced data that fits within the bounding box of the geostore with id provided. You can obtain the id of the geostore using the [RW API Geostore API](/index-rw.html#geostore). If the data is not geo-referenced, or if the dataset provider does not support the `geostore` query parameter, it will be ignored.

Reviewer: (/index-rw.html#geostore) -> (#geostore)

sql                    | The SQL query to be executed. This parameter changes the data returned in the query response body. | String | Yes |
format                 | The format of the returned response. By default, CSV format is assumed (`csv`), but you can also request the response as JSON (`json`). Check the section on the [download endpoint response body](index-rw.html#download-response-body) for some examples of how the `format` query parameter can be used. | String | No |
freeze                 | The `freeze` parameter, when provided as `true`, will create a file with the results of the execution of the query and return the URL for that file. **Please note that you should be authenticated in order to request freezing the results of query executions.** | Boolean | No |
geostore               | Read more about the geostore query parameter [here](/index-rw.html#filter-download-results-by-geostore). | String | No |

#### Filter download results by geostore

> Example download request providing a geostore id as query parameter to filter the results:

```shell

> curl -i -X GET 'http://api.resourcewatch.org/v1/query/1d7085f7-11c7-4eaf-a29a-5a4de57d010e?sql=SELECT * FROM dis_001_significant_earthquakes LIMIT 5&geostore=972c24e1da2c2baacc7572ee9501abdc'

Reviewer 2: Should this be a download request? Looks like a normal query request to me...

Author: Yes, it should! Thanks for pointing out :)

When downloading the query results for a document based dataset using either GET or POST `/download/<dataset id or slug>` request, the `FROM` clause is required but ignored, meaning you don't have to provide the dataset's `tableName` as you normally would. The example on the side illustrates this.

> ## Deleting data from a dataset

Reviewer: I'm not sure, but I suspect this process is asynchronous. Please test it, and if so, document its details, so users know that the dataset will go to "pending" state and they have to wait before using it.

Author: Done :)

> Example requests to delete data from a dataset:

```shell
curl -i -X GET 'https://api.resourcewatch.org/v1/query/:dataset_id?sql=DELETE FROM index_bf86b945c4ec41d2b5b7af00f3f61423'
curl -i -X GET 'https://api.resourcewatch.org/v1/query/:dataset_id?sql=DELETE FROM index_bf86b945c4ec41d2b5b7af00f3f61423 WHERE x = "y"'
```

Write queries such as `INSERT` or `UPDATE` are not supported in the RW API. You can use dataset endpoints to [append](/index-rw.html#concatenate-and-append-data-to-a-document-based-dataset) or [overwrite](/index-rw.html#overwrite-data-for-a-document-based-dataset) a given dataset's data, but you cannot use SQL to write data into the datasets.

> Most providers do not support `DELETE` queries as well. However, in the case of **document-based datasets**, you can delete the dataset's data via SQL `DELETE` query. Executing a `DELETE` query requires authentication, and additionally, one of the following conditions must be met:

Reviewer 2: ```suggestion
Most providers do not support `DELETE` queries either. However, in the case of **document-based datasets** (i.e. where the `connectorType` is `document`), you can delete the dataset's data via SQL `DELETE` query. Executing a `DELETE` query requires authentication, and additionally, one of the following conditions must be met:
```

| YES | Using PostGIS functions in WHERE clause | [SELECT \* FROM sp\_richness WHERE ST\_METADATA(the\_raster\_webmercator) IS NOT NULL LIMIT 5](http://api.resourcewatch.org/v1/query/16df8ada-87cc-4907-adce-a98bc4e91856?sql=SELECT%20\*%20FROM%20sp\_richness%20WHERE%20ST\_METADATA(the\_raster\_webmercator)%20IS%20NOT%20NULL%20LIMIT%205) |
| **NO** | **Using PostGIS functions in GROUP BY clause** | **[SELECT \* FROM sp\_richness GROUP BY ST\_SUMMARYSTATS(the\_raster\_webmercator, true) IS NOT NULL LIMIT 5](http://api.resourcewatch.org/v1/query/16df8ada-87cc-4907-adce-a98bc4e91856?sql=SELECT%20\*%20FROM%20sp\_richness%20GROUP%20BY%20ST\_SUMMARYSTATS(the\_raster\_webmercator%2C%20true)%20IS%20NOT%20NULL%20LIMIT%205)** |
| YES | Using PostGIS functions in ORDER BY clause | [SELECT \* FROM sp\_richness ORDER BY ST\_METADATA(the\_raster\_webmercator) IS NOT NULL LIMIT 5](http://api.resourcewatch.org/v1/query/16df8ada-87cc-4907-adce-a98bc4e91856?sql=SELECT%20\*%20FROM%20sp\_richness%20ORDER%20BY%20ST\_METADATA(the\_raster\_webmercator)%20IS%20NOT%20NULL%20LIMIT%205) |

> ### Arc-GIS Feature Service datasets

Reviewer: Arc-GIS -> ArcGIS

SQL-like queries can be employed for accessing data stored in Rasdaman datasets. Subsets on the original axes of the data may be provided in the WHERE statement. So far, only operations that result in a single scalar can be obtained from Rasdaman - averages, minimums, maximums.

```shell
curl -XGET 'https://api.resourcewatch.org/v1/query?sql=SELECT avg(Green) from 18c0b71d-2f55-4a45-9e5b-c35db3ebfe94 where Lat > 0 and  Lat < 45' \
  -H 'Content-Type: application/json' \
  -H 'Authorization: Bearer <token>'
```

### NEX-GDDP and Loca datasets

> Example of a query for a NEX-GDDP/Loca dataset, providing lat and lon as query parameters:

```shell
curl -X GET 'http://api.resourcewatch.org/v1/query/aaadd6c3-93ea-44bc-ba8b-7af3f40d39e1?sql=SELECT * FROM data&lat=30&lon=9'
```

> Example of a query for a NEX-GDDP/Loca dataset, providing a geostore id:

```shell

> curl -X GET 'http://api.resourcewatch.org/v1/query/aaadd6c3-93ea-44bc-ba8b-7af3f40d39e1?sql=SELECT * FROM data&geostore=972c24e1da2c2baacc7572ee9501abdc'

Reviewer: all example urls should use https

Author: Replaced all links using http with https :ok_hand:
